src/pages/Game/Game.js

```diff
@@ -8,77 +8,69 @@ import GameContent from "./GameContent";
 
 import {
   fetchQuestionsAndAnswers,
+  updateCorrectAnswer,
+  updateIncorrectAnswers,
   resetGame,
 } from "store/slices/gamePlaySlice";
 
 const Game = () => {
   const dispatch = useDispatch();
   const navigate = useNavigate();
   const {
-    categoryId,
-    difficulty,
-    answeredQuestionsCount,
     status,
-    questions,
     error,
+    answeredQuestionsCount,
+    difficulty,
+    questions,
+    categoryId,
   } = useSelector((state) => state.gamePlay);
 
   useEffect(() => {
     dispatch(fetchQuestionsAndAnswers({ categoryId, difficulty }));
   }, [categoryId, difficulty]);
 
+  useEffect(() => {
+    if (status === "succeeded" && questions.length !== 0) {
+      const currentQuestion = questions[answeredQuestionsCount];
+      dispatch(updateCorrectAnswer(currentQuestion.correct_answer));
+      dispatch(updateIncorrectAnswers(currentQuestion.incorrect_answers));
+    }
+  }, [answeredQuestionsCount, questions, status]);
+
   const handleAlertClose = () => {
     dispatch(resetGame());
     navigate("/");
   };
 
-  const content = () => {
-    switch (status) {
-      case "loading":
-        return <p>Loading...</p>;
-
-      case "failed":
-        return <p>Error: {error}</p>;
-
-      case "succeeded":
-        // Without the following, the code goes to case 'failed'
-        //Scenario when user has answered all questions correctly:
-        if (answeredQuestionsCount === questions.length) {
-          navigate("/gameover");
-        }
+  if (status === "loading") {
+    return <p>Loading...</p>;
+  }
 
-        if (
-          questions.length !== 0 &&
-          answeredQuestionsCount < questions.length
-        ) {
-          const currentQuestion = questions[answeredQuestionsCount];
+  if (status === "failed") {
+    return <p>Error: {error}</p>;
+  }
 
-          const shuffledAnswers = [
-            ...currentQuestion.incorrect_answers,
-            currentQuestion.correct_answer,
-          ].sort(() => Math.random() - 0.5);
-
-          return (
-            <GameContent
-              currentQuestion={currentQuestion}
-              answeredQuestionsCount={answeredQuestionsCount}
-              shuffledAnswers={shuffledAnswers}
-            />
-          );
-        }
-        return (
-          <AlertMessage
-            message={GAME_SCREEN_ALERTS.NO_QUESTIONS_MESSAGE}
-            title={GAME_SCREEN_ALERTS.NO_QUESTIONS_TITLE}
-            onClose={handleAlertClose}
-          />
-        );
-
-      default:
-        return null;
+  if (status === "succeeded") {
+    if (questions.length !== 0) {
+      const currentQuestion = questions[answeredQuestionsCount];
```

Comment:
This variable is passed as prop to the "GameContent" component and prop name is clear enough. So we can directly assign "questions[answeredQuestionsCount]"